src/plugins/intel_cpu/src/nodes/kernels/registers_pool.hpp

```diff
@@ -0,0 +1,368 @@
+        }
+        operator TReg&() { ensureValid(); return reg; }
+        operator const TReg&() const { ensureValid(); return reg; }
+        operator Xbyak::RegExp() const { ensureValid(); return reg; }
+        Reg& operator=(const StackAllocator::Address& addr) {
+            stack_mov(*this, addr);
+            return *this;
+        }
+        int getIdx() const { ensureValid(); return reg.getIdx(); }
+        friend Xbyak::RegExp operator+(const Reg& lhs, const Xbyak::RegExp& rhs) {
+            lhs.ensureValid();
+            return lhs.operator Xbyak::RegExp() + rhs;
+        }
+        void release() {
+            if (regPool) {
+                regPool->returnToPool(reg);
+                regPool.reset();
+            }
+        }
+        bool isInitialized() const { return static_cast<bool>(regPool); }
+
+    private:
+        void ensureValid() const {
+            if (!isInitialized()) {
+                IE_THROW() << "RegistersPool::Reg is either not initialized or released";
+            }
+        }
+
+        void initialize(const RegistersPool::Ptr& pool, int requestedIdx = anyIdx) {
+            static_assert(ov::util::IsAnyOf<TReg, Xbyak::Xmm, Xbyak::Ymm, Xbyak::Zmm,
+                                  Xbyak::Reg8, Xbyak::Reg16, Xbyak::Reg32, Xbyak::Reg64, Xbyak::Opmask>::value,
+                          "Unsupported TReg by RegistersPool::Reg. Please, use the following Xbyak registers either "
+                          "Reg8, Reg16, Reg32, Reg64, Xmm, Ymm, Zmm or Opmask");
+            release();
+            reg = TReg(pool->template getFree<TReg>(requestedIdx));
+            regPool = pool;
+        }
+
+    private:
+        TReg reg;
+        RegistersPool::Ptr regPool;
+    };
+
+    virtual ~RegistersPool() {
+        checkUniqueAndUpdate(false);
+    }
+
+    static Ptr create(x64::cpu_isa_t isa);
+    static Ptr create(x64::cpu_isa_t isa, std::initializer_list<Xbyak::Reg> regsToExclude);
+
+    template<typename TReg>
+    size_t countFree() const {
+        static_assert(ov::util::IsAnyOf<TReg, Xbyak::Xmm, Xbyak::Ymm, Xbyak::Zmm,
+                              Xbyak::Reg8, Xbyak::Reg16, Xbyak::Reg32, Xbyak::Reg64, Xbyak::Opmask>::value,
+                      "Unsupported TReg by RegistersPool::Reg. Please, use the following Xbyak registers either "
+                      "Reg8, Reg16, Reg32, Reg64, Xmm, Ymm, Zmm or Opmask");
+        if (std::is_base_of<Xbyak::Mmx, TReg>::value) {
+            return simdSet.countUnused();
+        } else if (std::is_same<TReg, Xbyak::Reg8>::value || std::is_same<TReg, Xbyak::Reg16>::value ||
+                   std::is_same<TReg, Xbyak::Reg32>::value || std::is_same<TReg, Xbyak::Reg64>::value) {
+            return generalSet.countUnused();
+        } else if (std::is_same<TReg, Xbyak::Opmask>::value) {
+            return countUnusedOpmask();
+        }
+    }
+
+    template<typename TReg>
+    inline TReg getInplaceFree(std::vector<Xbyak::Reg>& not_available) const {
```

Comment:
inline is redundant here

Reply:
This method was excluded from this PR along with uni_vgatherdps and uni_vscatterdps.